---
title: Dataset
---

## What dataset

We will use some samples from [a paper](https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000293) from Mark Pallen's group about comprehensive analysis of fecal metagenome samples from healthy individuals and patients with *Clostridium difficile* infection (CDI).

They found that CDI patients had significant alterations in their fecal microbiome composition, including a depletion of beneficial bacteria and an increase in potentially pathogenic species. The study also identified specific microbial signatures associated with CDI severity.

|Sample                                                           |#Seq              |Total bp     |
|:----------------------------------------------------------------|-----------------:|------------:|
|[SRX5707173](https://www.ncbi.nlm.nih.gov/sra/?term=SRX5707173)  |   **1,981,484**  |251,334,383  |
|[SRX5707285](https://www.ncbi.nlm.nih.gov/sra/?term=SRX5707285)  |   **4,534,703**  |552,669,038  |
|[SRX5707290](https://www.ncbi.nlm.nih.gov/sra/?term=SRX5707290)  |   **63,972**     |6,961,885    |
|[SRX5707359](https://www.ncbi.nlm.nih.gov/sra/?term=SRX5707359)  |   **561,807**    |62,895,743   |
|[SRX5707366](https://www.ncbi.nlm.nih.gov/sra/?term=SRX5707366)  |   **23,921,451** |2,832,669,901|
|[SRX5707374](https://www.ncbi.nlm.nih.gov/sra/?term=SRX5707374)  |   **10,304,335** |1,252,385,596|
|[SRX5707377](https://www.ncbi.nlm.nih.gov/sra/?term=SRX5707377)  |   **10,580,411** |1,279,917,137|

## Downloading the data

If you find an interesting dataset from a paper, it will be usually deposited to the [Sequence Read Archive](https://www.ncbi.nlm.nih.gov/sra) (SRA) at the National Center for Biotechnology Information (NCBI) or the [European Nucleotide Archive](https://www.ebi.ac.uk/ena) (ENA) at the European Bioinformatics Institute (EBI).

In both cases you will be able to determine a list of accession numbers, and using command line tools like:
* `fastq-dump` from the [SRA Toolkit](https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software)
* or a more complex pipeline like `nf-core/fetchngs` (see [docs](https://nf-co.re/fetchngs/))

You will be able to get the raw reads.
In our case, they will be already available in the `/shared/team/week5` folder.

## QC

When you get a new dataset is always worth performing a preliminary analysis (QC) to determine:

1. Number of reads (e.g. with `seqfu stats`)
2. Quality of the reads (e.g. with `fastqc` or `fastp`)
3. Potential contaminants (e.g. human reads)

## Subsampling

The dataset is quite large, and we will use a subset of reads to speed up the analysis.

```bash
# Example using seqkit
seqkit sample -p 0.2 SRX5707173_1.fastq.gz | gzip -c > SRX5707173_1.subsample.fastq.gz
```
