---
title: Blackboard
---

## :trophy: Badges

* Using **Anvi'o** to visualise metagenome assemblies (*in progress*)
* :medal_sports: Generating reports with **MultiQC** (Kraken2)
* :medal_sports: Taming NCBI Taxonomy with **TaxonKit**
* :medal_sports: Master of mapping with **Minimap2**

## :pencil: Tasks

#### Tuesday

* **:pencil2: Task1**: In `/shared/team/week5/reports` you will find four directories where we profiled the seven samples of our dataset with foud separate Kraken2 database. 
  * Choose 1 sample (ideally coordinate to survey them all) and create a directory with the four reports of the sample sample, addin a suffix to differentiate them.
  * Finally, run MultiQC to prepare a report using your input directory.

* **:pencil2: Task2**: Use a method described in our "Dataset" page to download `SRR27027485` (try, it's not a sample we will use)

* **:pencil2: Task3**: Use Krona and Pavian to analyse the Kraken reports of your favourite sample 

* **:pencil2: Task4**: Perform the de novo assembly of your favourite sample, after subsampling its reads to a maximum of 1.000.000.
  * See: https://mmbdtp.github.io/modules/metagenomics/assembly/
  * Use *fastp*  [is in the utils env] to perform a quality filtering, to retain only reads longer than 120 bp and with average quality of 19
  * Use *seqfu* *head* [same env] to select the first 1000000 reads of each file pair
  * Run megahit (you will need a new environment for this) to assemble it

## :bulb: Extras

#### :loop: Bash loops 

```bash
# Set a variable for convenience, then print its value (content)
DIR=/shared/team/week5/raw/fastq
echo $DIR

# Typical loop, the temporary variable is called "i"
for i in $DIR/*.gz;
do
  echo $i;
  echo "---- :) ----"
done
```

* Read [more on loops](https://telatin.github.io/microbiome-bioinformatics/Bash-tutorial-2/)