---
title: Reading list
---

## Learn more

1. [Anvi'o tutorials](https://anvio.org/learn/) - a fantastic curated list of tutorials based on Anvi'o from metagenomics to pan-genome analysis
   * :star: [Video: Basics of metagenomic read recruitment](https://www.youtube.com/watch?t=344&v=MqD4aN1p1qA&feature=youtu.be)
   * [Tutorial: Read recruitment](https://merenlab.org/tutorials/read-recruitment/) 
   * :star: [Video: Reconstructing genomes from  metagenomes](https://www.youtube.com/watch?v=RjNdHGK4ruo)
2. [bioBakery](https://huttenhower.sph.harvard.edu/tools/) - tools and tutorials for MetaPhlAn, HUMAnN, and other tools


## General overviews

1. Liu 2020, [A practical guide to amplicon and metagenomic analysis of microbiome data](https://link.springer.com/article/10.1007/s13238-020-00724-8): This review provides a comprehensive overview of current microbiome research methodologies, discussing the advantages and limitations of various tools and pipelines for amplicon and metagenomic analyses, and offers guidance on statistical and visualization methods, aiming to assist researchers in effective data analysis and tool selection for uncovering the biological significance of their data.
2. [A Metagenomics workflow](https://rpubs.com/ednachiang/MetaG_Pipeline) by Edna Chiang outlines a comprehensive metagenomic analysis pipeline, providing step-by-step instructions and recommendations for software and methods tailored to paired-end data analysis in metagenomics.
3. Yang 2021, [A review of computational tools for generating metagenome-assembled genomes from metagenomic sequencing data](https://www.sciencedirect.com/science/article/pii/S2001037021004931)

## Some methodological research papers

1. Delgado 2022, [Evaluating metagenomic assembly approaches for biome-specific gene catalogues](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-022-01259-2): advantages of mix-assembly
2. Mineeva 2020, [DeepMAsED: evaluating the quality of metagenomic assemblies](https://doi.org/10.1093/bioinformatics/btaa124): an example method for evaluating metagenomic assemblies.
